### DISCUSSION TITLE
Renault Zoe Q210; Unconfigured device, please add EOL settings

### DISCUSSION THREAD

**Discussion by LiquidSven** [cedricp/ddt4all#967] [Q&A] [answered]
I tried activating the rear camera in the MFD 5.1 for my Renault Zoe Q210. I have the base model of the car, and I've recently received a 6V camera to attacht to the back of the car, and saw a video of someone integrating it within the R-Link 1 system with success.

After connecting to the OBD port with the modified cable, the MFD 5.1 came up, I selected all the necessary options to 'Present', and then wrote them to the ECU.

I must've accidentally pressed the reset configuration button, because then I had the base settings for a Diesel Renault (Engine and gearbox type changed) and when I pressed the home button five times, the system showed this warning message.

I've read some articles of people who had the same problem, but they tried something with a tomtom.000 file, which I can't seem to download as the links have all expired.

I have tried to fill in some settings back to the right one, e.g. the gearbox and engine type back to electric, but I still have the yellow triangle showing unconfigured device, and telling me to add the EOL settings.

If anyone has any idea how to fix this (and if anyone knows if this is fixable), I'd gladly hear from you. I have an appointment at the dealership the 11th of January, but I'd rather fix this myself, as the dealership will probably cost me a lot of money. 

Thanks in advance!
![20231216_202108](https://github.com/cedricp/ddt4all/assets/62642068/e478798e-799c-4a84-872c-190e01f5c510)

**Comment by AndyIreland** [comment]
hello there - here is the tomtom.000 file for you I have uploaded here. You could give it a try and see if it works:
Make sure 'engine' is running throughout. It may take 20/30mins to load onto rlink so be patient. 
copy tomtom.000 to usb stick formatted to root of fat32 - if USB way dont work , try on a SD card instead (not the map SD card!!) 
https://we.tl/t-YSBqK63XEH

if this dont work and you have a bluetooth or USB ELM327 (not wi-fi) then if you want I could maybe get it going again remotely over the internet.
Good luck.

**Reply by LiquidSven** [reply]
Okay, thanks for the file. I'll give the tomtom.000 file a try. 

I'll get back to you!

**Reply by LiquidSven** [reply]
Sadly both the options didn't work. I both tried to start the car from blacked out with the USB and SD-card in it, and also tried to restart the system by pressing the home button 5 times, neither of these options worked.

The unconfigured device message kept reappearing instantly and nothing changed in the 25 minutes that I waited.

What could possibly be a next step?

**Reply by AndyIreland** [reply]
hard to tell unless I have a look myself and see what parameters are filled in and what are not - there could be some parameters not filled in or filled in incorrectly, I can only see if I log onto your rlink over internet.

**Reply by LiquidSven** [reply]
Okay, would you have time this saturday? I'm gone for work the next 2 days. Might it be a possibility for me to make a screenshot of all the different pages within the MFD 5.1 and send them here? Maybe that way you can see any problems.

If there are any people who have other solutions in the meantime, I'd gladly hear from them.

Thanks for the help thus far.

**Reply by AndyIreland** [reply]
you can try, I dont even know myself what I will be doing on saturday yet ... could be last minute christmas shopping who knows LOL . 
I am not a fan of that personally looking at screenshots, its not the same as me just rolling my sleves up and jumping in changing settings as I go along - its also more time consuming if I am looking at screenshots and telling you what to change instead of me doing it myself. 
Have you checked the VIN & CRC parameters have been written in on the MFD and radio 2012 ECU's ?

**Reply by LiquidSven** [reply]
I have checked the VIN, but what is supposed to be in the CRC parameters?

I'll just let you know when I have time then for checking the settings.

**Reply by AndyIreland** [reply]
Handily DDT4ALL has a built in CRC calculator ...... or you can just get the CRC detail from another ECU on your car
![image](https://github.com/cedricp/ddt4all/assets/49172217/41293273-0e88-438c-bf32-3d042af78906)

**Reply by LiquidSven** [reply]
Perfect, thanks. Is there anything else I should check aswell? Or is the VIN and CRC the most important box to check?

**Reply by AndyIreland** [reply]
well its one of the important things to make sure its filled out, but a lot more settings need filling out as well

**Reply by LiquidSven** [reply]
Alright, well thanks for the help so far. If you don't mind I'll contact you tomorrow or the day after for further assistance. I'll also let you know if I found anything out of the ordinary (CRC for example)

Really appreciate the help, thanks for taking the time!

**Reply by LiquidSven** [reply]
Okay, so I checked the VIN and CRC and they both check out.

Here are some photo's of the other pages within the MFD 5.1, maybe you can spot something out of the ordinary. I've checked most of them out, can't really find anything.
Does there need to be anything specific within the communication parameters?

![20231221_113927](https://github.com/cedricp/ddt4all/assets/62642068/a39068d9-f6e8-4629-9159-75d393bdd137)
![20231221_113936](https://github.com/cedricp/ddt4all/assets/62642068/abddf650-bb4e-4e76-a855-3b8ab9f5efaf)
![8c4cbda0-4676-4562-a432-c9b4953b5591](https://github.com/cedricp/ddt4all/assets/62642068/f4223a76-b42a-42d0-bb67-2a112d9372d7)

**Reply by AndyIreland** [reply]
most of them settings look OK - I would still need to see for myself remotely over the internet though to be sure and try some things

**Reply by LiquidSven** [reply]
Yes sure, if you don't mind.

I'll be back home tomorrow morning, at around 9AM my time (GMT+1), so I'll be available from then on. Just let me know how you'd like to set this up, and how you can connect remotely to my laptop!

Thanks again.

**Reply by AndyIreland** [reply]
I may be around from 2PM tomorrow UK time (if I am not out) - i use anydesk from anydesk.com to remotely hook up

**Reply by LiquidSven** [reply]
Alright, I'll send you a message tomorrow then. I'll make sure to get anydesk set up.

**Reply by AndyIreland** [reply]
Hi there - i am back now for a couple of hours if you want me to take a look at your settings and see if i can repair it. - if you still want me to all I need is your 9 digit anydesk ID number to log on.
Thanks,
Andy.

**Reply by LiquidSven** [reply]
I´ll get everthing connected to the car and then I´ll send you the code. Do you have a e-mail or number I can send it to? Might be a little better :)

**Comment by shrlnm** [comment]
@LiquidSven 
https://www.drive2.ru/b/643019689764615344/

**Reply by LiquidSven** [reply]
I've already read this post, but I didn't get pyren to work, and I'm also not in the possession of the Clip databases.

**Reply by AndyIreland** [reply]
try yivav68324@watrf.com

**Comment by AndyIreland** [comment]
> @LiquidSven https://www.drive2.ru/b/643019689764615344/

@shrlnm - Hi , i have just tried looking at the man's rlink remotely and alas we could not get it working for him - Everything looked in order all filled out (except for sound it looked) but all parameters looked filled out correctly.

I done:
Changing aftersales dates to todays date.
double checked the VIN and CRC were filled out correctly in MFD and radio and they were
used PyREN to clear DTC on MFD & Radio ecu's - MFD has 'alive' about missing configurations and is still alive after clearing - radio had no DTC
I was thinking of downloading a dump/txt backup file from my Zoe EV MFD & Radio which has exactly same rlink1 and change out the VIN & CRC details - but we did not know if that would work for sure or 'break it more' than it already is so he didnt go for it
Can you offer any more advice please?
Neither me or him can understand the details on Drive 2 you posted sorry about that - its a bit too technical

many thanks,
Andy.

**Reply by KarelSvo** [reply]
Will the dump of the current settings and the dump of your Zoe  be published?
So that more eyes can see and more minds can think?

**Reply by LiquidSven** [reply]
Can you tell me how I can achieve the dump from mine? Then maybe we can compare between mine and the one of Andy?
Or does it not work like that...

**Comment by AndyIreland** [comment]
> Will the dump of the current settings and the dump of your Zoe be published? So that more eyes can see and more minds can think?

Is there any implications or risks of uploading my dump file for my r-link1 MFD and RADIO ECU's  ? - if not I don't mind , but it will only be a .txt file as such with a load of numbers throughout it

**Reply by shrlnm** [reply]
@LiquidSven @AndyIreland 
I don't understand why you don't want to generate recovery macros for the Zoe LiquidSven VIN number.

Usually, when restoring RL1, the problem lies in the interaction of ELM and RL1. What type of adapter do you use? Is it USB BT or WIFI? I think that the result can only be achieved with a USB adapter.

**Reply by LiquidSven** [reply]
I used a USB adapter, we would be okay with using the macro as you described, but we are uncertain how to use the macro, and are unable to obtain the correct file.

Would you be able to assist?

**Comment by AndyIreland** [comment]
>  I don't understand why you don't want to generate recovery macros for the Zoe LiquidSven VIN number.
> 
> Usually, when restoring RL1, the problem lies in the interaction of ELM and RL1. What type of adapter do you use? Is it USB BT or WIFI? I think that the result can only be achieved with a USB adapter.

@shrlnm  I dont know how to make/generate Macro's from VIN - that's the thing. dont understand the instructions

**Reply by shrlnm** [reply]
@AndyIreland Just start a chat with telegram bot https://t.me/pyren_bot
then type "/start"
then type "/vin _your_vin_number_"
The bot sends you zip with macros

**Comment by KarelSvo** [comment]
@LiquidSven 
[Install software](https://transfiles.ru/p2rui), connect Elm interface, start software, select Elm connection, click MFD 5.4, write 1 and confirm with Enter key, write 15 and confirm with Enter key.
Go to the sendScripts/pyren/dumps folder.
Upload these files here.

**Reply by LiquidSven** [reply]
I'll get this going tomorrow, I'll get back to you. Thanks!

**Reply by LiquidSven** [reply]
[
[1703852169_11130.txt](https://github.com/cedricp/ddt4all/files/13793410/1703852169_11130.txt)
[1703852176_MFD_v5.4_20190228T163812.txt](https://github.com/cedricp/ddt4all/files/13793411/1703852176_MFD_v5.4_20190228T163812.txt)
](url)

Here you go @KarelSvo , here are the dumps. How would I go about inserting the macro´s into the radio? Feels like we're getting pretty close? Hopefully...

**Reply by KarelSvo** [reply]
Your MFD version does not correspond to the delivery status.
Take another dump of [your MID.](https://transfiles.ru/71dji)

**Reply by LiquidSven** [reply]
@KarelSvo, here are the dumps of the MID. Thanks for the help!

[1703941560_11129.txt](https://github.com/cedricp/ddt4all/files/13798103/1703941560_11129.txt)
[1703941574_Radio_2012_DAB_v4.0_20151012T163337.txt](https://github.com/cedricp/ddt4all/files/13798104/1703941574_Radio_2012_DAB_v4.0_20151012T163337.txt)

**Reply by KarelSvo** [reply, marked as answer]
[cmd.zip](https://github.com/cedricp/ddt4all/files/13802065/cmd.zip) Some parameters need to be checked and adjusted manually, good luck.

**Reply by LiquidSven** [reply]
@KarelSvo Do I just need to run those files with the ELM attached to the car. Or do I need to do anything else?

For now, happy new year and best wishes!

**Reply by KarelSvo** [reply]
https://github.com/cedricp/ddt4all/discussions/970#discussioncomment-7968378

**Reply by AndyIreland** [reply]
cant those cmd files just be added through the MACRO button in PyREN @KarelSvo  ?
![image](https://github.com/cedricp/ddt4all/assets/49172217/239283db-a4c7-4eda-b9ea-24fe70ac1b0b)

**Reply by KarelSvo** [reply]
@AndyIreland 
  yes the procedure is identical, but [LiquidSven](https://github.com/LiquidSven)   wrote           https://github.com/cedricp/ddt4all/discussions/967#discussioncomment-7910489 and https://github.com/cedricp/ddt4all/discussions/967#discussioncomment-7935921
                          I gave it an easy way to save the current state and current software reference. dtt4all does not create a dump file. The current MFD software version is not the delivery state. The MID has not been updated.

**Reply by KarelSvo** [reply]
@LiquidSven 
Check the MID front tweeter configuration.
They are configured in your dump. but the error memory says no connection to the front tweeter.
ECU : 11129  
Screen : DE (STD_B)
1  - DTC932113 (DF051) FRONT RIGHT TWEETER                                CO     PRESENT   
2  - DTC932213 (DF052) FRONT LEFT TWEETER                                 CO     PRESENT  
CO circuit open
PRESENT Error currently present

The VIN.CRC is missing in the MID and MFD.
Go to the ECM submenu
MFD command VP003
MID command VP004
Write VIN and confirm with Enter key
Write yes and confirm
Write your VIN and confirm.
The VIN.CRC is automatically calculated and added when sending.

**Reply by LiquidSven** [reply]
Okay I'll try this manually first.

Do you want me to also insert the macro's you've send me earlier? Via the 'Paired Interface Connection'?

Thanks!

**Reply by LiquidSven** [reply]
I´m going to write the VIN via the sendSCRIPTS.exe you´ve send me, but I can´t seem to find where I should insert the macro´s and cmd files you´ve send me (the MFD.cmd and MID.cmd). My pyREN.exe doesn't have a macro option in the main menu.
![afbeelding](https://github.com/cedricp/ddt4all/assets/62642068/18e36720-dbeb-4881-9c21-48497ef148c2)

**Reply by KarelSvo** [reply]
sendScripts is a PyRen version reduced to what is necessary.
This allows you to run cmd files. The macro key only has the name sendScript because ready-made scripts are sent.

**Reply by LiquidSven** [reply]
Where do I run the cmd files?

**Reply by KarelSvo** [reply]
Place in the program folder sendSCRIPTS/pyren/macro and select to send using the sendScript button.

**Reply by LiquidSven** [reply]
Okay, perfect I'm sending both the cmd files now.

Should I then hard reset my radio if both are finished?

**Reply by KarelSvo** [reply]
The hard reset command is included in both cmd files.

**Reply by LiquidSven** [reply]
Thanks so much!!!!!!! @KarelSvo and also @AndyIreland.

The MFD.cmd fixed it. It rebooted automatically and started right up, without any security codes needed or anything.

For people looking to fix it, I inserted the VIN into the ECM submenu in the sendSCRIPTS software and also inserted the script sent to me. The MID.cmd was not even necessary. 

Thanks again very much, can't be happier!!!

**Reply by KarelSvo** [reply]
You can change the *Unanswered* flag in the topic title.

**Reply by cardosoparis** [reply]
Hello @KarelSvo and @shrlnm. The pyren_bot does not respond and I need a recovery macro for a Renault Espace V R-Link 2/RadNav 3.X with a black screen after configuration corruption.
VIN: VF1RFC00260544918
Platform: XFC
Production date: 28.03.2018
Factory RadNav part number: 281154372R
Factory configuration reference: 283C21088R
Current ECU software reference: 281152288R
Current hardware reference: 285J99992R
RadNav family: 76
The ECU still communicates with DDT4All. Extended session 1003 works, but rollback was incomplete and several large configuration writes return “Service Not Supported In Active Session”. Many configuration parameters were reset to zero.
Could you please generate the correct PyRen recovery CMD/macros for this VIN and configuration, particularly the file corresponding to 283C21088R? I can provide the oldest and current DDT4All dumps.

**Reply by KarelSvo** [reply]
> The ECU still communicates with DDT4All. Extended session 1003 works, but rollback was incomplete and several large configuration writes return “Service Not Supported In Active Session”. Many configuration parameters were reset to zero.
> Could you please generate the correct PyRen recovery CMD/macros for this VIN and configuration, particularly the file corresponding to 283C21088R? I can provide the oldest and current DDT4All dumps.

ddt4all can write and load dumps? Since when?
Show the log files from your failed write attempt.

**Reply by KarelSvo** [reply]
@cardosoparis
The macroscript contains the configuration, including a description of what is being programmed. Please check it in a text editor and let me know if anything else needs to be changed:
[macro.zip](https://github.com/user-attachments/files/31053733/macro.zip)
  As already mentioned yesterday, the log files are necessary to rule out an interface problem.

**Reply by cardosoparis** [reply]
Thank you for your reply. I confirm that the file was executed by PyRen, not the extended session 1003 which responds positively (50 03:00 32 01 F4). The largest positive response is 6E, however the command 2E220301 returns 7F 2E 22 ("Condition Sequence Error"). The subsequent read 222203 returns 62 22 03 03.

Reset 1101 was also accepted (5101), but the screen and USB remain inactive. C DTC934149 - calculator configuration, active.

Current identification: part/software 281152288R, hardware 285.J99992R, reference 283C25279R, VIN VE1RFC00260544918.

I'm attaching the macro and the complete log. Could you check if an unlock/security condition is missing or if this state requires a complete firmware reprogramming?

**Reply by KarelSvo** [reply]
03222133
11C5622133FFFFFF
21FFFFFFFFFFFFFF
22FFFFFFFFFFFFFF
23FFFFFFFFFFFFFF
24FFFFFFFFFFFFFF
25FFFFFFFFFFFFFF
26FFFFFFFFFFFFFF
27FFFFFFFFFFFFFF
28FFFFFFFFFFFFFF
29FFFFFFFFFFFFFF
2AFFFFFFFFFFFFFF
2BFFFFFFFFFFFFFF
2CFFFFFFFFFFFFFF
2DFFFFFFFFFFFFFF
2EFFFFFFFFFFFFFF
2FFFFFFFFFFFFFFF
20FFFFFFFFFFFFFF
BUFFER FULL

Your China ELM interface gives up on this long byte string. I use a vLinker FS USB interface with a large buffer.

**Reply by cardosoparis** [reply]
Thanks. I understand: the answer of the 2133 identifier is approximately 453 bytes and my generic ELM ends with BUFFER FULL. I'm going to get a vLinker FS USB and repeat the read. When I have the new ecu_log.txt and elm_log.txt logs, I send it here before attempting any recording.

**Comment by LiquidSven** [comment]
> @AndyIreland Just start a chat with telegram bot https://t.me/pyren_bot then type "/start" then type "/vin _your_vin_number_" The bot sends you zip with macros

@shrlnm the bot doesn't work for me, could you possibly make the macro? VIN is VF1AGVYA052169097

**Reply by KarelSvo** [reply]
[cmd](https://transfiles.ru/esz59)
Place these two cmd files in the macro folder of the sendSCRIPTS software that I gave you and load your recovery data into the two radio control units.
Requirement: The radio firmware version must still be original.

**Reply by LiquidSven** [reply]
Thanks a lot for the help.

I'll shoot you a message if I find any difficulties, was busy today with Christmas and family etc.

Will try it tomorrow afternoon.

**Reply by LiquidSven** [reply]
@KarelSvo 
Hi, I'm currently in the sendSCRIPTS program, but when I open the program and click on MFD 5.4, I only get the following screen. I've inserted the macro's you send me in the sendSCRIPTS/pyren/macro's folder.
![17038504043457851700158286244089](https://github.com/cedricp/ddt4all/assets/62642068/a9ab7ec9-94a2-4ad0-86c6-e0cd7942d9e3)

**Reply by LiquidSven** [reply]
![17038504762617140615114643837945](https://github.com/cedricp/ddt4all/assets/62642068/a4be7c68-49f2-4060-9333-fa8a36d5ecc2)

It also says this after I press 1 (R-Link 1)

**Reply by KarelSvo** [reply]
Sorry, an error in the mod_utils.py file.
Exchange the file.
[mod_utils.py.zip](https://github.com/cedricp/ddt4all/files/13793392/mod_utils.py.zip)

**Comment by AndyIreland** [comment]
> > @AndyIreland Just start a chat with telegram bot https://t.me/pyren_bot then type "/start" then type "/vin _your_vin_number_" The bot sends you zip with macros

I don't fancy giving my phone number details on telegram - any other way to do it?

**Comment by LiquidSven** [comment]
I just found this in the After Sales data in DDT4ALL, in the Diagnosis part.

It says in there that the engine type is Diesel and vehicle type is B98, is there a way to change this? This might be the reason it's not booting?
![20231229_131325](https://github.com/cedricp/ddt4all/assets/62642068/38b242bb-aa76-4f67-89a1-a15778c50215)

**Comment by AndyIreland** [comment]
> Thanks so much!!!!!!! @KarelSvo and also @AndyIreland.
> 
> The MFD.cmd fixed it. It rebooted automatically and started right up, without any security codes needed or anything.

Excellent , so glad you finally got it fixed - well done to @KarelSvo without his help I dont think you would have made it

**Comment by KarelSvo** [comment]
> Just start a chat with telegram bot https://t.me/pyren_bot
> then type "/start"
> then type "/vin _your_vin_number_"
> The bot sends you zip with macros

I noticed the following in the Liquidsven example.

The MFD part number in BVM database is 3034R, with reference number 0805R.
/vin _your_vin_number_ extracts a reference script 6696R.cmd

These references are assigned to part number 3034R in the BVM X10 references file.
0805R
4883R
2447R
7187R*
2764R
6696R*
6245R
0949R
5406R
4577R
9999R
On the Reprog DVD there are only recovery files with the numbers marked with a *.
If there is no direct match the reference number, does the pyren_bot always choose the recovery file with the highest reference number? Or are other selection parameters used?

The currently read MFD part number is 5104R.
/vin [your_vin_number]  [current part number]  extracts also a reference script 6696R.cmd
The BVM X10 reference file gives the reference number 9349R for this part number.
There is a recovery file for reference number 9349R on the reprog DVD.
Why doesn't the pyren_bot send a 9349R.cmd file in response to this request?

**Comment by Ras-srv** [comment]
Good Morning everyone,

Need help from the experts about bringing the R-link on my 2016 Zoe back to life.
I was trying to set the clock on the R-Link but the option was greyed out for some reason. So I connected my Launch scanner and decided to do a reset to the R-link thinking it will bring it back to factory settings. But it wiped all the settings and now it's displaying unconfigured radio message.

Any help is much appreciated, I have downloaded and setup DDt4all, but had no luck so far bringing it to life. 
Any experts in here who can help?

I see from the chat history here some deep diving into scripts and macros.

**Reply by Ras-srv** [reply]
its a 2016 Renault Zoe Q210 rapid charge model
![IMG_4353](https://github.com/user-attachments/assets/2bda19f6-77ae-4523-8d59-a125b7c3c1d3)

**Comment by AndyIreland** [comment]
@Ras-srv -   You could try this fix that  doesnt involve using DDT4ALL. Download from here

https://we.tl/t-82dUCs5Hk1

 (file expires in a couple of days) and put the tomtom.000 file onto an 8GB USB stick thats been formatted to Fat32 and put it in usb socket of car and keep Zoe in 'ready' mode so the radio doesnt turn off. if nothing looks like  its happening press the 'home' button next to the screen 5 times to reboot the rlink radio. 
If it does not work with the USB way copy the tomtom.000 file onto the root of a 8gb SD card formatted to FAT32 (DO NOT USE THE MAP SD CARD THAT IS IN THE ZOE USE ANOTHER ONE)

See if that works. if not it will be DDT4ALL way with a compatible ELM327 and a modified splitter cable needed and all the parameters to be put back into the rlink 

good luck.

**Reply by Ras-srv** [reply]
Hello Andy. 
Thank you for your response. 
I will give it a go tonight when I get home. 
I do have a Proscan branded ELM327 that is compatible with DTT4ALL. I use it to reset the charging impossible error I get every time I charge to 100%. That's my next problem to fix once the R-Link is back to life. 

Do I still need a modified splitter cable to talk to the R-link unit?

Thanks. 
Firas.

**Reply by AndyIreland** [reply]
you do normally need a modified splitter cable (so I am predicting it the same applies with your Proscan ELM327 ?)  - because you are actually reading/writing to the rlink that works through the CAN2 or CAN-H or CAN-HI network through the OBD2 socket. 

Here is how you modify it/wire it up:
https://www.gps-rlink.com/r-link-1/activate-android-auto-waze-on-r-link-1/

**Reply by Ras-srv** [reply]
Hello Andy. 

I haven't had any luck unfortunately with the tomtom file and USB flash. 

I have purchased and modified a OBD2 cable and confirm can access the head unit. 

What do you think the next step would be?

**Reply by AndyIreland** [reply]
to see if every parameter is filled out. - if you get stuck and you got the bluetooth or usb elm327 and a windows laptop i can log on over internet and have a look if you want and see if I can see what the problem is. you more than likely will need the security code for the rlink when you restore the settings, normally found at back of owners manual or off the 'my renault' website

**Reply by Ras-srv** [reply]
Great, Thank you.
I will have a look this weekend and send you some screen shots.

**Reply by Ras-srv** [reply]
how is your availability in case I need remote help?

**Reply by AndyIreland** [reply]
i am in and out this weekend but if you send me a private message when you are set up and ready to go I will have a look and see if I can do it for you. 
email: andyr<@>gofree.indigo.ie

**Reply by Ras-srv** [reply]
Great. Thank you. 

I haven't got any documentation that came with the car. So will need to see how to get from the website.

**Reply by AndyIreland** [reply]
fill out your vin and add your car to the my renault app and it will have your radio code listed there (if its the original rlink that came with the car!)

![image](https://github.com/user-attachments/assets/ba47027c-c47b-41ce-a090-c994baad23a6)

**Comment by Renotech1** [comment]
If it and r link 1 you dont need to modify the can H L . We can just enter
read write with ddt and elm327.. if r link 2 / Easylink we need bypass
gateway or modify Can network

Op ma 2 jun 2025, 20:23 schreef Andy Robus ***@***.***>:

**Reply by AndyIreland** [reply]
I cant agree - to read and fill in parameters with DDT4ALL of the Radio2012 and MFD you need to either modify a ELM327 by soldering , or get a splitter lead and make modifications same with r-link2 . 
Later cars have easyconnect unit and the later renaults have security gateway on OBD connector that has to be unlocked with token or gateway bypassed.

**Reply by paulotoze** [reply]
Hello everyone. I need help with a similar situation to the ones described. I have the famous yellow triangle saying radio not configured. I don't know which version of R-link I have. The VIN is VF1AGVYA150121714. Can you help me?

**Reply by AndyIreland** [reply]
> Hello everyone. I need help with a similar situation to the ones described. I have the famous yellow triangle saying radio not configured. I don't know which version of R-link I have. The VIN is VF1AGVYA150121714. Can you help me?

you have a Renault Zoe ZE 22kw with rlink1 radio then - you need the ELM327 and a specially wired cable to read rlink through OBD2 socket and fill in parameters of the rlink and radio2012

**Reply by paulotoze** [reply]
Hi Andy, I already have the modified obd wired cable in order to read CAN2 and the ELM327. but I don't know the next steps. Can you help?

**Reply by AndyIreland** [reply]
i can but its bedtime LOL  - i will post the details tomorrow for you

**Comment by Renotech1** [comment]
Mine can 😀 and this is older version. Mine have also Search function with
vehicles ecu type. + auto scan that works. + better layout response + all
functions that renolink 2.14 have (add key / reset airbag ect ect)



Op do 13 aug 2026, 22:09 schreef KarelSvo ***@***.***>:

**Comment by cardosoparis** [comment]
Hello @KarelSvo and @shrlnm. The pyren_bot does not respond and I need a recovery macro for a Renault Espace V R-Link 2/RadNav 3.X with a black screen after configuration corruption.
VIN: VF1RFC00260544918
Platform: XFC
Production date: 28.03.2018
Factory RadNav part number: 281154372R
Factory configuration reference: 283C21088R
Current ECU software reference: 281152288R
Current hardware reference: 285J99992R
RadNav family: 76
The ECU still communicates with DDT4All. Extended session 1003 works, but rollback was incomplete and several large configuration writes return “Service Not Supported In Active Session”. Many configuration parameters were reset to zero.
Could you please generate the correct PyRen recovery CMD/macros for this VIN and configuration, particularly the file corresponding to 283C21088R? I can provide the oldest and current DDT4All dumps.

**Reply by mishu28nmv** [reply]
Here you go: 
[VF1RFC00260544918.zip](https://github.com/user-attachments/files/31132914/VF1RFC00260544918.zip)

**Comment by cardosoparis** [comment]
Merci beaucoup Je vais essayer lá 😇😇😇😇

Château de Montigny

Marcos CARDOSO

1680 Route de Nouziere

Dissay - 86130

☎️:
0623031678

📧:
***@***.***
-------- Message d'origine --------De : KarelSvo ***@***.***> Date : 14/08/2026  04:34  (GMT+01:00) À : cedricp/ddt4all ***@***.***> Cc : cardosoparis ***@***.***>, Mention ***@***.***> Objet : Re: [cedricp/ddt4all] Renault Zoe Q210; Unconfigured device, please add EOL settings (Discussion #967) 
@cardosoparis
The macroscript contains the configuration, including a description of what is being programmed. Please check it in a text editor and let me know if anything else needs to be changed:
macro.zip

**Reply by cardosoparis** [reply]
[ecu_log.txt](https://github.com/user-attachments/files/31078585/ecu_log.txt)
[elm_log.txt](https://github.com/user-attachments/files/31078586/elm_log.txt)
